src/Domain/Infra/DependencyInjection/Compiler/DoctrineObjectFieldMappingPass.php

```diff
@@ -39,7 +39,7 @@ public function process(ContainerBuilder $container): void
                 throw new InvalidArgumentException(sprintf('Provider "%s" must implement "%s".', $providerId, ObjectFieldMappingsProviderInterface::class));
             }
 
-            foreach ($providerClass::provideObjectFieldMappings() as $class => $mapping) {
+            foreach ($providerClass::provideObjectFieldMappings($container->getParameter('msgphp.doctrine.mapping.key_max_length')) as $class => $mapping) {
```

Comment:
i like to move this to runtime, so we inject all tagged providers in `ObjectFieldMappingListener`.

then we can pass the `MappingConfig` for an argument, and each providers receives all available configs